tools/VersionController/Models/VersionBumper.cs

```diff
@@ -0,0 +1,348 @@
+        /// </summary>
+        public void BumpAllVersions()
+        {
+            var moduleName = _fileHelper.ModuleName;
+
+            Console.WriteLine("Bumping version for " + moduleName + "...");
+
+            using (PowerShell powershell = PowerShell.Create())
+            {
+                powershell.AddScript("$metadata = Test-ModuleManifest -Path " + _fileHelper.OutputModuleManifestPath + ";$metadata.Version;$metadata.PrivateData.PSData.Prerelease");
+                var cmdletResult = powershell.Invoke();
+                _oldVersion = cmdletResult[0]?.ToString();
+                _isPreview = !string.IsNullOrEmpty(cmdletResult[1]?.ToString());
+            }
+
+            if (_oldVersion == null)
+            {
+                throw new Exception("Unable to obtain old version of " + moduleName + " using the built module manifest.");
+            }
+
+            _newVersion = IsNewModule() ? _oldVersion : GetBumpedVersion();
+
+            if (_oldVersion == _newVersion)
+            {
+                Console.WriteLine(_fileHelper.ModuleName + " is a new module. Keeping the version at " + _oldVersion);
+
+                // Generate the serialized module metadata file
+                _metadataHelper.SerializeModule();
+            }
+            else
+            {
+                Console.WriteLine("Updating version for " + _fileHelper.ModuleName + " from " + _oldVersion + " to " + _newVersion);
+            }
+
+            UpdateSerializedAssemblyVersion();
+            UpdateChangeLog();
+            var releaseNotes = GetReleaseNotes();
+            UpdateOutputModuleManifest(releaseNotes);
+            UpdateRollupModuleManifest();
+            UpdateAssemblyInfo();
+            UpdateDependentModules();
+
+            Console.WriteLine("Finished bumping version " + moduleName + "\n");
+        }
+
+        /// <summary>
+        /// Apply a version bump to a given version.
+        /// </summary>
+        /// <returns>The updated version after the bump has been applied.</returns>
+        private string GetBumpedVersion()
+        {
+            var moduleName = _fileHelper.ModuleName;
+            var splitVersion = _oldVersion.Split('.').Select(v => int.Parse(v)).ToArray();
+            var versionBump = _metadataHelper.GetVersionBumpUsingSerialized();
+            if (string.Equals(moduleName, "AzureRM.Profile"))
+            {
+                var commonCodeVersionBump = _metadataHelper.GetVersionBumpForCommonCode();
+                if (commonCodeVersionBump == Version.MAJOR)
+                {
+                    versionBump = Version.MAJOR;
+                }
+                else if (commonCodeVersionBump == Version.MINOR && versionBump == Version.PATCH)
+                {
+                    versionBump = Version.MINOR;
+                }
+            }
+
+            // PATCH update for preview modules (0.x.x or x.x.x-preview)
+            if (splitVersion[0] == 0 || _isPreview)
```

Comment:
let's just set VersionBump to Revision in tis case and use the code below to set tthe actual version